app/controllers/ApplicationController.scala

```diff
@@ -60,8 +59,8 @@ trait ApplicationController extends BaseController {
   }
 
   def applicationStatus(applicationId: String) = Action.async { implicit request =>
-    appRepository.findProgress(applicationId).map { result =>
-      Ok(Json.toJson(ApplicationStatusOrder.getStatus(result)))
+    appRepository.findStatus(applicationId).map { result =>
+      Ok(Json.toJson(result))
     }.recover {
       case e: ApplicationNotFound => NotFound(s"cannot find application for user with id: ${e.id}")
```

Comment:
Not sure if the exception message is appropriate because what we want is the application status, so we fail and we say application not found, that might be confusing. I would say cannot retrieve applications status details for application with id=$applicationId. Instead of the userId, I think applicationId is more relevant in this case.